lib/paths.bzl

```diff
@@ -195,6 +195,30 @@ def _relativize(path, start):
   result_segments = segments[-length:]
   return "/".join(result_segments)
 
+def _relative_path(target, start):
+  """Returns a relative path to `target` from `start`.
+
+  Args:
+    target: path that we want to get relative path to.
+    start: path to directory from which we are starting.
```

Comment:
I guess the assumption is both paths are rooted at the same place to start with?

Should it check for getting a relative path in either arg?

This is pretty close to relativize, the docs are going to need to call out exactly what is different and why (assuming it is worth having both).

Reply:
The function is completely different. `paths.relativize` requires the path you want to relativize to contain the other argument as its prefix. Here we do a more general thing: we tell how to get to `target` path starting from `start` path. `paths.relativize` hasn't been very useful to us, while the new function is quite handy.

Reply:
A better path forward then would be to simply extend the existing function to handle that scenario. Adding a new function with a similar name will simply lead to confusion.

Reply:
Sometimes you don't want to use absolute paths, e.g. when writing rpaths.

Reply:
I suggest using relativize and adding a keyword argument to specify this .. behavior is desired. Otherwise it might break people who assume relativize fails under certain scenarios.